platform/src/common/components/Collocation/AddMonitor/Table/DataTable.jsx

```diff
@@ -21,7 +21,7 @@ const DataTable = ({ paginatedData, collocationDevices }) => {
 
   const handleSelectAllDevices = (e) => {
     const allDevices = [];
-    collocationDevices.map((device) => allDevices.push(device._id));
+    collocationDevices.map((device) => allDevices.push(device.device));
```

Comment:
Device has a property called device?
This does not look right.
Could we stick to deviceID or another property rather than this?